### PR TITLE
Decode spark-local mounts as mounts

### LINKED ISSUE #30: Spark local mounts are decoded as directories.

### PR DESCRIPTION
**Pull request by beckyconning**
Fixes #30

### CODE REVIEW

**Comment by beckyconning** [comment]
@cryogenian please review.

File: src/Quasar/FS/Mount.purs [hunk 1]
```diff
@@ -53,7 +55,8 @@ fromJSON parent = decodeJson >=> \obj → do
     "directory", "mongodb" → Right $ MongoDB (parent </> dir name)
     "directory", "couchbase" → Right $ Couchbase (parent </> dir name)
     "directory", "marklogic" → Right $ MarkLogic (parent </> dir name)
-    "directory", "spark" → Right $ SparkHDFS (parent </> dir name)
+    "directory", "spark-hdfs" → Right $ SparkHDFS (parent </> dir name)
+    "directory", "spark-local" → Right $ SparkHDFS (parent </> dir name)
```
Comment (garyb) [inline src/Quasar/FS/Mount.purs]: This is mapping to `SparkHDFS` rather than `SparkLocal`

File: src/Quasar/FS/Resource.purs [hunk 2]
```diff
@@ -42,6 +42,8 @@ instance showResource ∷ Show Resource where
   show (Directory p) = "(Directory " <> show p <> ")"
   show (Mount m) = "(Mount " <> show m <> ")"
 
+-- This only produces Directory or File. Is this what we want?
+-- See also `Quasar.FS.Mount.fromJson`.
```
Comment (garyb) [inline src/Quasar/FS/Resource.purs]: ??

```
Mount <$> Mount.fromJSON parent json -- <-- mount
<|> do
  -- file-or-directory stuff
```

**Comment by garyb** [comment]
Stealing the review ;)

File: src/Quasar/FS/Mount.purs (hunk 1 above)
Comment (beckyconning) [inline src/Quasar/FS/Mount.purs]: ayyy lmao

File: src/Quasar/FS/Resource.purs (hunk 2 above)
Comment (beckyconning) [reply src/Quasar/FS/Resource.purs]: `u r correct`

Comment (garyb) [reply src/Quasar/FS/Resource.purs]: This is good to go once you delete the comment :wink:

**Review by garyb — CHANGES_REQUESTED** [review]

**Review by garyb — APPROVED** [review]

**Pull request merged**